Source/utils/SmallDataIO.cpp

```diff
@@ -188,3 +299,45 @@ void SmallDataIO::get_specific_data_line(std::vector<double> &a_out_data,
     std::vector<double> coords(1, a_coord);
     get_specific_data_line(a_out_data, coords);
 }
+
+// ------------ Other Functions --------------
+
+std::string SmallDataIO::get_new_filename(const std::string &a_file_prefix,
+                                          double a_dt, double a_time,
+                                          const std::string &a_file_extension,
+                                          int a_filename_steps_width)
+{
+    const int step = std::round(a_time / a_dt);
```

Comment:
Could you add an assertion or mayday error for dt!=0 ?
For example in the Puncture Tracker Refactor PR (#122 ), for time=0 I set dt=1, which I did after getting myself into trouble for first trying dt=0. An assertion would be good.